app/queries/district_analytics_query.rb

```diff
@@ -34,6 +35,22 @@ def results
     results.inject(&:deep_merge)
   end
 
+  def total_assigned_patients
+    @total_assigned_patients ||=
+      Patient
+        .with_hypertension
+        .joins(:assigned_facility)
```

Comment:
We don't need to join the full facilities table here. We should be able to do:
`.where(assigned_facility: facilities).group(:assigned_facility_id)`

Reply:
TBH, I was just imitating the total_registered_patients method below. Your suggestion does seem to yield the same results. Thank you!

Reply:
Ha! Room for optimization everywhere 😂